### PR TITLE
refactor(cli): DRY cli/ast.rs

### PR DESCRIPTION
**Pull request by bartlomieju**

### CODE REVIEW

File: cli/tests/bundle/fixture15.out [hunk 1]
```diff
@@ -1,3 +1,3 @@
 export function getIndex(c) {
-    return "\x00\r\n\x85\u2028\u2029".indexOf(c);
+    return "\u{0}\r\n\u{85}\u{2028}\u{2029}".indexOf(c);
```
Comment (kitsonk) [inline cli/tests/bundle/fixture15.out:2]: Something is wrong if you need to do this now... @kdy1 fixed this, so it shouldn't modify this...

**Review by kitsonk — COMMENTED** [review]
Something isn't right if you are changing fixture15... it isn't supposed to change at all.

File: cli/tests/bundle/fixture15.out (hunk 1 above)
Comment (kitsonk) [reply cli/tests/bundle/fixture15.out:2]: See: https://github.com/denoland/deno/pull/8162#issuecomment-731719973

Comment (bartlomieju) [reply cli/tests/bundle/fixture15.out:2]: Thanks for pointing this out, you're right that it shouldn't change, that said, I'm a bit lost on the cause of this change, removed code is exactly the same as code in `parse` 😬 I'll look for the problem

Comment (bartlomieju) [reply cli/tests/bundle/fixture15.out:2]: Actually, it seems that `transpile_module` already passes `Rc<SourceMap>` as an argument

**Comment by bartlomieju** [comment]
Closing without merge for now

File: cli/tests/bundle/fixture15.out (hunk 1 above)
Comment (bartlomieju) [reply cli/tests/bundle/fixture15.out:2]: Alright, I've managed to fix it by splitting `parse()` into `parse_module()` (that works as `parse()` did) and adding lower-level utility called `parse_with_source_map()`

**Review by kitsonk — APPROVED** [review]
LGTM

**Pull request merged**